middleware/src/permutation.rs

```diff
@@ -0,0 +1,20 @@
+use crate::circuit::ColumnMid;
+
+// TODO: Dedup with other Cell definition, or move this to a higher level
```

Comment:
Can't be deduped easily because the other `Cell` has region info, and this one is in the middleware where the concept or Region doesn't apply.  I'll move it to a higher level instead.

Reply:
Addressed in https://github.com/privacy-scaling-explorations/halo2/pull/254/commits/9927c824927cf82acf7a7c45848a7f86f85a7416